File: Python/LinearDataStructures/Stack/Stack.py
class Stack:

    def __init__(self, size):
        self.size = size
        self.stack = []
        self.top = -1
    
    #? Push/Insert an element in the stack
    def push(self, data):
        if self.size-self.top>1:
            self.top +=1
            self.stack.append(data)
        else:
            print("Stack Overflow!")
    
    #? Popped/Delete the top element of the stack
    def pop(self):
        if self.top==-1:
            print("Stack Underflow!")
        else:
            self.stack.pop()

    #? Display elements present in the stack
    def show(self):
        print("Stack: ")
        for i in self.stack[::-1]:
            print(i)
    
    #? Number of elements present in the stack 
    def count(self):
        value = len(self.stack)
        return value

s = Stack(5)
s.push(1)
s.push(2)
s.push(3)
s.show()
print("Count: ",s.count())
s.pop()
s.show()
print("Count: ",s.count())

File: CDev/stack.c
#include <stdio.h>
#define MAX 5

int stack[MAX];
int top=-1;

void push(int data){
    if(top==MAX-1){
        printf("Stack Overflow\n");
        return;
    }
    
    top+=1;
    stack[top] = data;
}

void pop(){
    if(top==-1){
        printf("Stack underflow");
        return;
    }

    printf("Removed element is: %d", stack[top]);
    top+=1;
}

void peek(){
    if(top==-1){
        printf("Stack underflow");
        return;
    }

    printf("Top element: %d", stack[top]);
}

int size(){
    return top+1;
}

void display(){
    for(int i=top; i>=0; i--){
        printf("%d\n", stack[i]);
    }
}

int main(){
    push(1);
    push(2);
    push(3);

    display();
    int s = size();
    printf("Size of stack : %d", size());
    return 0;
}